### PR TITLE
chart atoms

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?
Renders Chart Atoms!

The styling is a bit off. The height of the Chart isn't known so wrapping in a `u-responsive-ratio` div to show the entire chart... and a massive bit of whitespace...

## Screenshots
image

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/views/fragments/atoms/atom.scala.html [hunk 1]
```diff
@@ -22,6 +23,7 @@
         case profile: ProfileAtom => views.html.fragments.atoms.snippets.profile(profile, isAmp = amp)
         case explainer: ExplainerAtom if !amp => Html(ArticleAtomRenderer.getHTML(explainer.atom, config))
         case commonsdivision: CommonsDivisionAtom if !amp => Html(ArticleAtomRenderer.getHTML(commonsdivision.atom, config))
+        case chart: ChartAtom if !amp => views.html.fragments.atoms.chart(chart, config)
```
Comment (regiskuckaertz) [inline common/app/views/fragments/atoms/atom.scala.html]: You should be using the atom rendering library instead here.

File: common/app/views/fragments/atoms/chart.scala.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@import com.gu.contentatom.renderer.{ArticleAtomRenderer, ArticleConfiguration}
+
+@(chart: model.content.ChartAtom, config: ArticleConfiguration)
+
+<div class="u-responsive-ratio">
+    @Html(ArticleAtomRenderer.getHTML(chart.atom, config))
```
Comment (regiskuckaertz) [inline common/app/views/fragments/atoms/chart.scala.html]: Oh, I see. Great 👍

Comment (akash1810) [reply common/app/views/fragments/atoms/chart.scala.html]: Yeah, I'm trying to get the iframe's body to be the correct height. Wrapping it in `u-responsive-ratio` gets somewhere close, but results in a lot of whitespace (see screenshot).

Comment (regiskuckaertz) [reply common/app/views/fragments/atoms/chart.scala.html]: It is unfortunately really hard to do. The next solution would be to use web components 😎

**Comment by gtrufitt** [comment]
Interactive atoms use a [resize listener](interactiveResize) I think, can we use similar concept to avoid whitespace?

**Comment by gtrufitt** [comment]
> So just need to work out how to get frontend to listen to the message...

Wrap it up in interactive atom template? https://github.com/guardian/frontend/blob/master/common/app/views/fragments/atoms/interactive.scala.html

**Comment by regiskuckaertz** [comment]
We've done it in a separate PR, so closing this one.

**Pull request closed without merging**